routers/web/repo/view.go

```diff
@@ -999,10 +999,18 @@ func renderCode(ctx *context.Context) {
 			ctx.ServerError("GetBaseRepo", err)
 			return
 		}
-		ctx.Data["RecentlyPushedNewBranches"], err = git_model.FindRecentlyPushedNewBranches(ctx, ctx.Repo.Repository.ID, ctx.Doer.ID, ctx.Repo.Repository.DefaultBranch)
-		if err != nil {
-			ctx.ServerError("GetRecentlyPushedBranches", err)
-			return
+
+		showRecentlyPushedNewBranches := true
+		if ctx.Repo.Repository.IsMirror ||
+			!ctx.Repo.Repository.UnitEnabled(ctx, unit_model.TypePullRequests) {
```

Comment:
I think this unit check should be `baseRepo.UnitEnabled()` not `ctx.Repo.UnitEnabled()`.
If the context repo is a fork repo without PR unit, we should also show them as the base repo's PR unit is enabled and we can create PRs.
And for the similar reason, we should also check whether baseRepo is a mirror repo.

Will be fixed in #25812.

Reply:
And maybe we also need to check the unit permission. 🤔